### PR TITLE
Accept selector instead of function name

### LINKED ISSUE #1783: Accept selector instead of function name
### Which components does the task require to be changed? (think hard pls)

`snforge` + `sncast`

### Description

Change functions to accept selector instead of name.

Current
```cairo
start_mock_call(contract_address, 'my_function', ret_data);
```

New
```rust
start_mock_call(contract_address, selector!("my_function"), ret_data);
```

### PR DESCRIPTION
**Pull request by Draggu**
Closes #1783

## Introduced changes

- snforge
  - `start_mock_call()`
  - `stop_mock_call()`
  - `L1Handler`
- sncast
  - `call()`
  - `invoke()`
  
Now uses selector `selector!()` instead of names.

## Checklist

- [x] Linked relevant issue
- [x] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [x] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/rpc.rs [hunk 1]
```diff
@@ -200,10 +201,10 @@ pub fn call_l1_handler(
     syscall_handler: &mut SyscallHintProcessor,
     runtime_state: &mut RuntimeState,
     contract_address: &ContractAddress,
-    entry_point_selector: &Felt252,
+    entry_point_selector: Felt252,
     calldata: &[Felt252],
 ) -> CallResult {
-    let entry_point_selector = create_entry_point_selector(entry_point_selector);
+    let entry_point_selector = EntryPointSelector(entry_point_selector.into_());
```
Comment (piotmag769) [inline crates/cheatnet/src/runtime_extensions/call_to_blockifier_runtime_extension/rpc.rs:207]: Unrelated change

File: docs/src/appendix/cheatcodes/l1_handler_execute.md [hunk 2]
```diff
@@ -26,7 +26,7 @@ struct L1Handler {
 where:
 
 - `contract_address` - The target contract address
-- `function_name` - Name of the `#[l1_handler]` function
+- `function_selector` - Name of the `#[l1_handler]` function
```
Comment (piotmag769) [inline docs/src/appendix/cheatcodes/l1_handler_execute.md:29]:
```suggestion
- `function_selector` - Selector of the `#[l1_handler]` function
```

File: docs/src/appendix/cheatcodes.md [hunk 3]
```diff
@@ -12,8 +12,8 @@
 - [`start_spoof`](cheatcodes/start_spoof.md) - changes the transaction context for contracts
 - [`stop_spoof`](cheatcodes/stop_spoof.md) - cancels the `start_spoof` for contracts
 - [`get_class_hash`](cheatcodes/get_class_hash.md) - retrieves a class hash of a contract
-- [`start_mock_call`](cheatcodes/start_mock_call.md) - mocks contract call to a `function_name` of a contract
-- [`stop_mock_call`](cheatcodes/stop_mock_call.md) - cancels the `start_mock_call` for the function `function_name` of a contract
+- [`start_mock_call`](cheatcodes/start_mock_call.md) - mocks contract call to a `function_selector` of a contract
+- [`stop_mock_call`](cheatcodes/stop_mock_call.md) - cancels the `start_mock_call` for the function `function_selector` of a contract
```
Comment (piotmag769) [inline docs/src/appendix/cheatcodes.md:16]:
```suggestion
- [`start_mock_call`](cheatcodes/start_mock_call.md) - mocks contract call to a function selected with `function_selector` of a contract
- [`stop_mock_call`](cheatcodes/stop_mock_call.md) - cancels the `start_mock_call` for the function selected with `function_selector` of a contract
```

and in other places too

File: docs/src/appendix/cheatcodes/stop_mock_call.md [hunk 4]
```diff
@@ -14,7 +14,7 @@ use snforge_std::stop_mock_call;
 fn test_mock_call() {
     // ...
     
-    stop_mock_call(contract_address, function_name);
+    stop_mock_call(contract_address, function_selector);
```
Comment (drknzz) [inline docs/src/appendix/cheatcodes/stop_mock_call.md:17]: Not so useful of an example :D
Let's at least show the selector! macro in use
```suggestion
    stop_mock_call(contract_address, selector!("my_function"));
```

File: docs/src/appendix/sncast-library/call.md [hunk 5]
```diff
@@ -14,7 +14,7 @@ pub struct CallResult {
 ```
 
 - `contract_address` - address of the contract to call.
-- `function_name` - the name of the function to call, as Cairo shortstring.
+- `function_selector` - the selector of the function to call, as Cairo shortstring.
```
Comment (drknzz) [inline docs/src/appendix/sncast-library/call.md:17]: It's no longer cairo shortstring

File: docs/src/appendix/sncast-library/invoke.md [hunk 6]
```diff
@@ -18,7 +18,7 @@ pub struct InvokeResult {
 ```
 
 - `contract_address` - address of the contract to invoke.
-- `entry_point_selector` - the name of the function to invoke, as Cairo shortstring.
+- `entry_point_selector` - the selector of the function to invoke, as Cairo shortstring.
```
Comment (drknzz) [inline docs/src/appendix/sncast-library/invoke.md:21]: Not shortstring

**Review by THenry14 — COMMENTED** [review]
let's fix sncast tests as well

**Review by abulenok — APPROVED** [review]

File: crates/cheatnet/src/runtime_extensions/common.rs [hunk 7]
```diff
@@ -12,5 +12,5 @@ pub fn create_execute_calldata(calldata: &[Felt252]) -> Calldata {
 
 #[must_use]
 pub fn create_entry_point_selector(entry_point_selector: &Felt252) -> EntryPointSelector {
-    EntryPointSelector(entry_point_selector.clone().into_())
+    EntryPointSelector(entry_point_selector.clone().into_()) //FIXME: remove clone and take `entry_point_selector` by value
```
Comment (war-in) [inline crates/cheatnet/src/runtime_extensions/common.rs:15]: 👀

**Review by war-in — APPROVED** [review]
It's fine on the forge side 🥇

**Review by THenry14 — APPROVED** [review]
